### DISCUSSION TITLE
Example for AddLinqToDBContext<TContext, TContextImplementation>

### DISCUSSION THREAD

**Discussion by s-kotte** [linq2db/linq2db#4324] [Q&A] [answered]
I'm trying to use the `AddLinqToDBContext<TContext, TContextImplementation>` method in my ASP.NET Core application to resolve an interface instead of the concrete class in my controllers. I am getting the following error message:

> Unable to resolve service for type 'LinqToDB.DataOptions`1[Identity.SetupConfiguration.DB.DbConnection]' while attempting to activate 'Identity.SetupConfiguration.DB.DbConnection'.

My DbConnection class:
```c#
public class DbConnection: DataConnection, IDbConnection
{
    public DbConnection(DataOptions<DbConnection> options) : base(options.Options)
    {
    }

    public ITable<UserDb> User => this.GetTable<UserDb>();
}

public interface IDbConnection: IDataContext
{
    public ITable<UserDb> User { get; }
}
```

I've also registered the DataConnection in Startup.cs like this:
```c#
builder.Services.AddLinqToDBContext<IDbConnection, DbConnection>((provider, options) => options
    .UsePostgreSQL(builder.Configuration.GetConnectionString("Default")!)
    .UseDefaultLogging(provider));
```

**Comment by MaceWindu** [comment]
Change `public DbConnection(DataOptions<DbConnection> options)` to `public DbConnection(DataOptions<IDbConnection> options)`

**Reply by s-kotte** [reply]
Thanks, do I need to wait until issue #4326  is fixed or am I still doing something wrong? I now get the error Missing constructor accepting 'DataOptions' on type DbConnection.

**Reply by MaceWindu** [reply, marked as answer]
Add second constructor for now:
```cs
public DbConnection(DataOptions<IDbConnection> options) : base(options.Options)
{
}
public DbConnection(DataOptions<DbConnection> options) : base(options.Options)
{
}
```
issue will be fixed in next release

**Comment by MaceWindu** [comment]
@linq2db, any idea why [we use](https://github.com/linq2db/linq2db/blob/6bb5d8baaff795e44a38af7dc63019b38a346652/Source/LinqToDB.AspNet/ServiceConfigurationExtensions.cs#L164) `TContext` instead of `TContextImplementation` ?

**Reply by MaceWindu** [reply]
That's actually bug... We pass `TContextImplementation` to `HasTypedContextConstructor` but use `TContext` for registration

**Reply by MaceWindu** [reply]
Filled issue #4326, should be an easy fix
